pepper-apis/src/test/java/org/broadinstitute/ddp/db/dao/UserGovernanceDaoTest.java

```diff
@@ -6,18 +6,39 @@
 import static org.junit.Assert.assertNull;
 import static org.junit.Assert.assertTrue;
 
+import java.util.Collection;
+import java.util.List;
+
 import org.broadinstitute.ddp.TxnAwareBaseTest;
+import org.broadinstitute.ddp.cache.DaoBuilder;
 import org.broadinstitute.ddp.db.TransactionWrapper;
 import org.broadinstitute.ddp.model.governance.Governance;
 import org.broadinstitute.ddp.model.user.User;
 import org.broadinstitute.ddp.util.TestDataSetupUtil;
 import org.junit.BeforeClass;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
 
+@RunWith(Parameterized.class)
 public class UserGovernanceDaoTest extends TxnAwareBaseTest {
+    private DaoBuilder<UserGovernanceDao> daoBuilder;
+    private boolean isCachedDao;
 
     private static TestDataSetupUtil.GeneratedTestData testData;
 
+    @Parameterized.Parameters
+    public static Collection<Object[]> data() {
+        Object[] uncached = {(DaoBuilder<UserGovernanceDao>)(handle) -> handle.attach(UserGovernanceDao.class), false};
+        Object[] cached = {(DaoBuilder<UserGovernanceDao>)(handle) -> new UserGovernanceCachedDao(handle), true};
+        return List.of(uncached, cached);
+    }
+
+    public UserGovernanceDaoTest(DaoBuilder daoBuilder, boolean isCachedDao) {
+        this.daoBuilder = daoBuilder;
+        this.isCachedDao = isCachedDao;
+    }
+
```

Comment:
Was trying to figure out how I could test `CachedDao` versions without duplicating all the tests. This is one solution I found: JUnit will instantiate and run the test for each element in the method annotated with `@Parameterized.Parameters` . So here we can run the same tests, once with the regular implementation and the other with the uncached. The DaoBuilder interface equalizes instantiation of JdbiDao and CachedDaos.